icechunk/src/format/manifest.rs

```diff
@@ -175,21 +176,25 @@ impl FromIterator<ChunkInfo> for Manifest {
 struct PayloadIterator {
     manifest: Arc<Manifest>,
     for_node: NodeId,
+    ndim: usize,
     last_key: Option<(NodeId, ChunkIndices)>,
 }
 
 impl Iterator for PayloadIterator {
     type Item = (ChunkIndices, ChunkPayload);
 
     fn next(&mut self) -> Option<Self::Item> {
+        // TODO: tie the MAX to the type in ChunkIndices.
+        let upper_bound =
+            ChunkIndices(itertools::repeat_n(u32::MAX, self.ndim).collect());
         match &self.last_key {
             None => {
                 if let Some((k @ (_, coord), payload)) = self
                     .manifest
                     .chunks
                     .range((
                         Bound::Included((self.for_node.clone(), ChunkIndices(vec![]))),
-                        Bound::Unbounded,
+                        Bound::Included((self.for_node.clone(), upper_bound)),
```

Comment:
Why is `clone` not needed on `upper_bound`? Is it because both the `Bound` and `upper_bound` go out of scope at the same time?

Reply:
upper_bound can transfer ownership. Rust is smart enough to know that only one branch of the match  will execute, so you can move the variable. `self.for_node` on the other hand cannot, because it would partially destroy `self`.